### ISSUE TITLE
TLS/SSL support not available; install glib-networking

### ISSUE THREAD

**Issue by khinsen** [atlas-engineer/next#592]
**Describe the bug**
When I try to open a HTTPS URL, I get the error message "TLS/SSL support not available; install glib-networking".

**Precise recipe to reproduce the issue**
- Start `next`
- Open any HTTPS URL

**Information**
- OS name+version: Guix commit 1c9e3e9f36621bb91d153bc155c6cbce704e9220
- Kernel name+version:
- Graphics card and driver:
- Desktop environment / Window manager name+version: None. I ssh into Guix from macOS running XQuartz.
- Next version (from =M-x next-version= or =next --version=): commit e3e3d2bfa003f0b1d4e428b1d6e9a2eff9129b99
- How you installed Next (Guix pack, package manager, build from source): I run Next from source without installation
- SBCL version (if built from source): 2.0.2
- WebKitGTK+ / PyQt-Webengine version:

**Comment by jmercouris** [comment]
Thank you for reporting this issue! I would also just like to let you know that you can install Next very easily on macOS now, in fact I am typing this coment from there :-]
<img width="1036" alt="Screen Shot 2020-03-07 at 22 55 44" src="https://user-images.githubusercontent.com/1691662/76153039-cb5ecc00-60c6-11ea-95b7-dc4f041eca64.png">

**Comment by Ambrevar** [comment]
@khinsen You need to install glib-networking to your user profile.
This should be done with the Guix package.  If it does not work, we need to move it to propagated inputs.

**Comment by jmercouris** [comment]
I know what you mean, C and M work right now, as well as shift, I don’t know about Hyper or Super as I haven’t tested those very thoroughly. If you are interested in getting it to work, I can update the instructions on macOS

**Comment by Ambrevar** [comment]
@khinsen: Do you have gsettings-desktop-schemas in your profile as well?

Which OS do you use?  Is it Guix System?

**Comment by gravitational-dark-light** [comment]
make sure you have set GIO_MODULE_DIR after installing glib-networking

it didn't work for me until I did that

**Comment by Ambrevar** [comment]
If you didn't install Next to a default profile, or if you are on a
foreign distribution, make sure you sourced etc/profile.

The GIO_* env variables might be exposed by the glib* packages.

**Comment by khinsen** [comment]
Thanks everyone, I have made some progress:
- Installing `glib-networking` and/or `gsettings-desktop-schemas` in my profile makes no difference. Next still doesn't work with https, and there are no GIO* environment variables in my profile.
- After setting `export GIO_MODULE_DIR=$HOME/.guix-profile/lib/gio/modules`, Next (current git checkout, run from source) works fine.
- When I install Next into my profile (using `build-scripts/guix.scm`), I have exactly the same error as when running from source, but setting `GIO_MODULE` doesn't fix it.

Summary: I have absolutely no idea what's going on, but I have a workaround for running Next.

**Comment by jellelicht** [comment]
Installing `gsettings-desktop-schemas` and `glib-networking` into my profile only made a difference after either setting the `GIO_MODULE_DIR` myself, or simply reloading my profile by restarting my session.

**Comment by Ambrevar** [comment]
I run Guix system and I don't have GIO_MODULE_DIR in my environment.
Next,  gsettings-desktop-schemas and glib-networking are all installed
into profiles (not even the default profile).

I am a bit puzzled at this point... :p

**Comment by khinsen** [comment]
Do you run a desktop on your Guix system? I don't. It's a virtual machine running headless. If your setup has something like a full Gnome installation, that could explain the difference.

**Comment by Ambrevar** [comment]
I run EXWM.

**Comment by khinsen** [comment]
After browsing the GLib documentation (in particular https://developer.gnome.org/gio/2.62/running-gio-apps.html) and the Guix source code, it looks like Guix doesn't use `GIO_MODULE_DIR`, but rather `GIO_EXTRA_MODULES`. That looks indeed more appropriate because it is a list of directories, so it can be composed piecewise for a profile. Here's an appropriate comment from `glib-or-gtk-build-system`:
```
;; All GIO modules are expected to be installed in GLib's $libdir/gio/modules
;; directory.  That directory has to include a file called giomodule.cache
;; listing all available modules.  GIO can be made aware of modules in other
;; directories with the help of the environment variable GIO_EXTRA_MODULES.
;; The official GIO documentation states that this environment variable should
;; only be used for testing and not in a production environment.  However, it
;; appears that there is no other way of specifying multiple modules
;; directories (NIXOS also does use this variable). See
;; https://developer.gnome.org/gio/stable/running-gio-apps.html
```

Apparently something goes wrong there, but I don't claim to understand this in any detail.

**Comment by Ambrevar** [comment]
Indeed, I have

```
GIO_EXTRA_MODULES=/home/ambrevar/.guix-extra-profiles/dev/dev/lib/gio/modules:/gnu/store/3x6cbx98m05kq6505x7r5chwx72m5mrb-dconf-0.32.0/lib/gio/modules
```

**Comment by khinsen** [comment]
So it's `dconf`. If I install `dconf` to my profile, I get `GIO_EXTRA_MODULES` set and Next works just fine. And if I add `dconf` to the inputs for Next, it works fine as well, without any extra efforts. Pull request coming.

**Comment by Ambrevar** [comment]
Fixed in ebd569858479af709f58f71a349449846de647eb.
